core/types/src/report.rs

```diff
@@ -2,15 +2,35 @@
 //! SGX Report
 
 use crate::{
-    impl_newtype_for_bytestruct, new_type_accessors_impls, svn::ConfigId, Attributes, ConfigSvn,
-    CpuSvn, IsvSvn, Measurement, MiscellaneousSelect, MrEnclave, MrSigner,
+    impl_newtype_for_bytestruct, key_request::KeyId, new_type_accessors_impls, svn::ConfigId,
+    Attributes, ConfigSvn, CpuSvn, IsvSvn, Measurement, MiscellaneousSelect, MrEnclave, MrSigner,
 };
 use mc_sgx_core_sys_types::{
-    sgx_isvext_prod_id_t, sgx_isvfamily_id_t, sgx_prod_id_t, sgx_report_body_t, sgx_report_data_t,
-    SGX_ISVEXT_PROD_ID_SIZE, SGX_REPORT_BODY_RESERVED1_BYTES, SGX_REPORT_BODY_RESERVED2_BYTES,
+    sgx_isvext_prod_id_t, sgx_isvfamily_id_t, sgx_mac_t, sgx_prod_id_t, sgx_report_body_t,
+    sgx_report_data_t, sgx_report_t, SGX_ISVEXT_PROD_ID_SIZE, SGX_MAC_SIZE,
+    SGX_REPORT_BODY_RESERVED1_BYTES, SGX_REPORT_BODY_RESERVED2_BYTES,
     SGX_REPORT_BODY_RESERVED3_BYTES, SGX_REPORT_BODY_RESERVED4_BYTES, SGX_REPORT_DATA_SIZE,
 };
 
+/// MAC
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct Mac(sgx_mac_t);
+
+new_type_accessors_impls! {
+    Mac, sgx_mac_t;
+}
+
+impl Mac {
+    pub const SIZE: usize = SGX_MAC_SIZE;
```

Comment:
Same Q as the other PR: should this be public, or can we keep it private?

Reply:
Removed it. Along with being able to derive Default it was over exposing API members thinking users might need to know how to construct a key. At this time not seeing a need for a user to write or generate a MAC, only to verify one.